pkgs/development/compilers/gcc/default.nix

```diff
@@ -408,14 +409,28 @@ lib.pipe ((callFile ./common/builder.nix {}) ({
       maintainers
     ;
   } // lib.optionalAttrs (!atLeast11) {
-    badPlatforms = if !is49 then [ "aarch64-darwin" ] else lib.platforms.darwin;
+    badPlatforms = if !(is48 || is49) then [ "aarch64-darwin" ] else lib.platforms.darwin;
   };
-} // optionalAttrs is7 {
-  env.NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.cc.isClang && langFortran) "-Wno-unused-command-line-argument";
+} // optionalAttrs (!atLeast11 && stdenv.cc.isClang) {
+  env.NIX_CFLAGS_COMPILE = toString (
+    lib.optionals (lib.versionAtLeast (lib.getVersion stdenv.cc) "16") [
+      # Silence errors related to use of implicitly declared library functions and implicit ints,
+      # which can cause unwanted `configure` feature detection failures.
+      "-Wno-error=implicit-function-declaration"
+      "-Wno-error=implicit-int"
+    ] ++ lib.optionals (is7 && langFortran) [
+      "-Wno-unused-command-line-argument"
+    ]
+  );
+} // lib.optionalAttrs (!atLeast10 && stdenv.hostPlatform.isDarwin) {
+  # GCC <10 requires default cctools `strip` instead of `llvm-strip` used by Darwin bintools.
+  preBuild = ''
+    makeFlagsArray+=('STRIP=${lib.getBin darwin.cctools-port}/bin/${stdenv.cc.targetPrefix}strip')
+  '';
 } // optionalAttrs (!atLeast7) {
   env.langJava = langJava;
 } // optionalAttrs atLeast6 {
-  NIX_LDFLAGS = lib.optionalString  hostPlatform.isSunOS "-lm";
+  env.NIX_LDFLAGS = lib.optionalString hostPlatform.isSunOS "-lm";
```

Comment:
This change might be a good idea, but it doesn't need to go to `staging-next`.  It also doesn't belong in a Darwin-specific PR.

Reply:
See
- https://github.com/NixOS/nixpkgs/pull/266193

Reply:
This was requested by @wegank. If there’s already work being done on it, I’ll just drop the commit.